### PR TITLE
Feature crop borders

### PR DESCRIPTION
**Pull request by axiel7**
- Added to Manga reader only (idk if webtoons would benefit from this)
- Seems to work fine on white backgrounds and simple black backgrounds
- Added as a setting, off by default

Feel free to request any code changes!

### CODE REVIEW

**Comment by axiel7** [comment]
It was lagging when fast scrolling pages so I used the same approach as downsampling and added a new processor

File: iOS/UI/Reader/Page/CropBordersProcessor.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+//
+
+import Foundation
+import Nuke
+
+#if os(iOS) || os(tvOS)
+import UIKit
+#else
+import CoreGraphics
+#endif
```
Comment (Skittyblock) [inline iOS/UI/Reader/Page/CropBordersProcessor.swift:15]: you only need to import coregraphics
```suggestion
import CoreGraphics
```

**Comment by Skittyblock** [comment]
seems to work pretty good, nice! the only thing I'd want is to have it refresh the pages after disabling/enabling the setting, as currently it only takes effect on pages that haven't been loaded yet.

as a side note, the same thing should be done for the downsample images option (but I forgot), but you don't need to include that in this pr.

File: iOS/UI/Reader/Page/ReaderPageView.swift [hunk 2]
```diff
@@ -167,6 +172,13 @@ class ReaderPageView: UIView {
                         image = processedImage
                     }
                 }
+                if UserDefaults.standard.bool(forKey: "Reader.cropBorders") {
+                    let processor = CropBordersProcessor()
+                    let processedImage = processor.process(image)
+                    if let processedImage = processedImage {
```
Comment (Skittyblock) [inline iOS/UI/Reader/Page/ReaderPageView.swift:178]: for future reference (I'll add this to the swiftlint rules in the future as I need to change it throughout the app)
```suggestion
                    if let processedImage {
```

**Review by Skittyblock — APPROVED** [review]

**Pull request merged**